addon/webextension/background/selectorLoader.js

```diff
@@ -81,6 +81,15 @@ this.selectorLoader = (function() {
     });
   };
 
+  function incognitoCheck(tabId) {
+    return browser.tabs.get(tabId).then(tab => {
+      return browser.tabs.executeScript(tabId, {
+        code: `window.incognito = ${tab.incognito};`,
+        runAt: "document_start"
+      });
+    });
+  }
```

Comment:
I think this can be:

```js

function incognitoCheck(tabId) {
  return browser.tabs.get(tabId).then(tab => {
    return tab.incognito;
  });
}
```

See also [tabs.Tab](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/tabs/Tab)

Reply:
Ah, actually this function has to assign `window.incognito` inside the content script context, so that the ui.js scripts can detect private browsing mode. The window global is the content script global, which is [different](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Content_scripts#Content_script_environment) from the background page global and from the underlying website's global.